core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java

```diff
@@ -288,48 +287,37 @@ public static long diffDate(@SqlType("varchar(x)") Slice unit, @SqlType(Standard
         return getDateField(UTC_CHRONOLOGY, unit).getDifferenceAsLong(DAYS.toMillis(date2), DAYS.toMillis(date1));
     }
 
-    private static DateTimeField getDateField(ISOChronology chronology, Slice unit)
-    {
-        String unitString = unit.toStringUtf8().toLowerCase(ENGLISH);
-        switch (unitString) {
-            case "day":
-                return chronology.dayOfMonth();
-            case "week":
-                return chronology.weekOfWeekyear();
-            case "month":
-                return chronology.monthOfYear();
-            case "quarter":
-                return QUARTER_OF_YEAR.getField(chronology);
-            case "year":
-                return chronology.year();
-        }
-        throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "'" + unitString + "' is not a valid DATE field");
-    }
-
-    public static DateTimeField getTimestampField(ISOChronology chronology, Slice unit)
-    {
-        String unitString = unit.toStringUtf8().toLowerCase(ENGLISH);
-        switch (unitString) {
-            case "millisecond":
-                return chronology.millisOfSecond();
-            case "second":
-                return chronology.secondOfMinute();
-            case "minute":
-                return chronology.minuteOfHour();
-            case "hour":
-                return chronology.hourOfDay();
-            case "day":
-                return chronology.dayOfMonth();
-            case "week":
-                return chronology.weekOfWeekyear();
-            case "month":
-                return chronology.monthOfYear();
-            case "quarter":
-                return QUARTER_OF_YEAR.getField(chronology);
-            case "year":
-                return chronology.year();
-        }
-        throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "'" + unitString + "' is not a valid Timestamp field");
+    private static DateTimeField getDateField(ISOChronology chronology, Slice unitString)
+    {
+        DateTimeUnit unit = DateTimeUnit.valueOf(unitString)
+                .orElseThrow(() -> new TrinoException(INVALID_FUNCTION_ARGUMENT, "'" + unitString.toStringUtf8() + "' is not a valid DATE field"));
+
+        return switch (unit) {
+            case DAY -> chronology.dayOfMonth();
+            case WEEK -> chronology.weekOfWeekyear();
+            case MONTH -> chronology.monthOfYear();
+            case QUARTER -> QUARTER_OF_YEAR.getField(chronology);
+            case YEAR -> chronology.year();
+            default -> throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "'" + unit + "' is not a supported DATE field");
```

Comment:
Is the error message different here on purpose? If it is, what's the difference? Can we phrase it better, so it's more clear?